experimental/ggml/benchmark_ggml.sh

```diff
@@ -0,0 +1,110 @@
+#!/bin/bash
+#
+# Copyright 2023 The OpenXLA Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# Environment variables:
+# PYTHON: Python interpreter, default: /usr/bin/python3
+# OOBI_VENV_DIR: path to create Python virtualenv, default: ggml-benchmarks.venv
+# OOBI_TARGET_DEVICE: target benchmark device, can also be specified the first
+#   argument.
+# OOBI_BUILD_DIR: path to the GGMl build directory.
+# OOBI_OUTPUT: path to output benchmark results, can also be specified the
+#   second argument.
+#
+# Example usage:
+# ./benchmark_ggml.sh <target-device> <build-dir> <result-path>
+
+set -xeuo pipefail
+
+VENV_DIR="${OOBI_VENV_DIR:-ggml-benchmarks.venv}"
+PYTHON="${PYTHON:-/usr/bin/python3}"
+TARGET_DEVICE_NAME="${1:-${OOBI_TARGET_DEVICE}}"
+BUILD_DIR="${2:-${OOBI_BUILD_DIR}}"
+OUTPUT_PATH="${3:-${OOBI_OUTPUT}}"
+
+TD="$(cd $(dirname $0) && pwd)"
+
+# Setup virtual environment.
+VENV_DIR="${VENV_DIR}" PYTHON="${PYTHON}" source "${TD}/setup_venv.sh"
+
+# Initialize results json.
+OUTPUT_PATH="$(realpath ${OUTPUT_PATH})"
+"${TD}/../../comparative_benchmark/scripts/create_results_json.sh" "${OUTPUT_PATH}"
+
+pushd "${BUILD_DIR}"
+
+PROMPT="Once upon a time"
+BENCHMARK_BINARY="$(realpath bin/gpt-2)"
+WARMUP_ITERAIONS=2
+NUM_ITERATIONS=10
+
+MODEL="$(realpath models/gpt-2-117M/ggml-model-f32.bin)"
+
+declare -a BENCHMARK_NAMES=(
+  "models/GPT2LMHEAD_FP32_GGML/inputs/INPUT_DATA_MODEL_DEFAULT"
+  "models/GPT2LMHEAD_FP16_GGML/inputs/INPUT_DATA_MODEL_DEFAULT"
+  "models/GPT2LMHEAD_INT4_GGML/inputs/INPUT_DATA_MODEL_DEFAULT"
+)
+declare -a MODELS=(
+  ggml-model-f32.bin
+  ggml-model-f16.bin
+  ggml-model-q4_0.bin
+)
+declare -a DATA_TYPES=(
+  "fp32"
+  "fp16"
+  "int4"
+)
+
+declare -a args=(
+  --warmup_iterations "${WARMUP_ITERAIONS}"
+  --iterations "${NUM_ITERATIONS}"
+  --benchmark_binary "${BENCHMARK_BINARY}"
+  --prompt "${PROMPT}"
+  --seed 0
+  --output "${OUTPUT_PATH}"
+  --target_device "${TARGET_DEVICE_NAME}"
+  --verbose
+)
+
+if [[ "${TARGET_DEVICE_NAME}" =~ ^(pixel-4|pixel-6-pro|moto-edge-x30)$ ]]; then
+BENCHMARK_SCRIPT="run_benchmarks_android.py"
```

Comment:
nit: Maybe have one indent as these are in a if block? Same on the else block

Reply:
Updated.